kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -315,6 +325,37 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
     executor.submit(new RunnableReadTask(new ReadTaskState(task, state, callback)));
   }
 
+  class ReadFutureTask<V> extends FutureTask<V> {
+
+    private Runnable myTask;
+
+    public ReadFutureTask(Runnable runnable, V result) {
+      super(runnable, result);
+      this.myTask = runnable;
+    }
+
+    public Runnable getReadTask() {
+      return myTask;
+    }
+  }
+
+  class KafkaConsumerThreadPoolExecutor extends ThreadPoolExecutor {
+    public KafkaConsumerThreadPoolExecutor(int corePoolSize,
+                                               int maximumPoolSize,
+                                               long keepAliveTime,
+                                               TimeUnit unit,
+                                               BlockingQueue<Runnable> workQueue,
+                                               RejectedExecutionHandler handler) {
+    super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, handler);
+
+    }
+
+    @Override
+    protected <T> RunnableFuture<T> newTaskFor(Runnable runnable, T value) {
+      return new ReadFutureTask(runnable, value);
```

Comment:
We only want to wrap `RunnableReadTask` in `ReadFutureTask` since that's the class that requires special handling.
We could do it similar to this:
```
if (runnable instanceof RunnableReadTask) {
      return new ReadFutureTask(runnable, value);
}
return super.newTaskFor(runnable, value);
```

Reply:
@huangyiminghappy I think we need to do this as we don't always call `executor.submit()` on `RunnableReadTask`